### DISCUSSION TITLE
is interleave search deterministic between platforms?

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#4441] [CP-SAT questions] [answered]
Hello,
  I found difference in progress log between macos and other OSes (window10, ubuntu, debian).

  Or-tools version is 9.10 and interleave search is used.

Starting deterministic search at 30.05s with 1 workers and batch size of 0.
9 full problem subsolvers: [core, default_lp, fixed, max_lp, no_lp, pseudo_costs, quick_restart, quick_restart_no_lp, reduced_costs]
10 incomplete subsolvers: [graph_arc_lns, graph_cst_lns, graph_dec_lns, graph_var_lns, rnd_cst_lns, rnd_var_lns, scheduling_intervals_lns, scheduling_precedences_lns, scheduling_resource_windows_lns, scheduling_time_window_lns]
3 helper subsolvers: [neighborhood_helper, synchronization_agent, update_gap_integral]
Setting number of tasks in each batch of interleaved search to 1

  Macos finds following bounds:
#Bound 126.17s best:inf   next:[10,600]   fixed
#Bound 1212.80s best:inf   next:[14,600]   max_lp

 Other OSes on various hardware(epyc, threadripper, ryzen) finds following bounds:
#Bound 141.29s best:inf   next:[10,600]   fixed
#Bound 1381.50s best:inf   next:[12,600]   fixed
#Bound 3022.28s best:inf   next:[14,600]   max_lp

   Lower bound 12 is not found in macos and compared to other cpus, bound 14 is found very quickly. I attached result for 7950x. Results from macos are measured on macbook pro with m2 pro cpu.

  Is it an error and on every hardware and os should be found bounds and solutions same or is m2 pro lucky for my model and due to fast unified memory can find next bound much earlier?

Thanks,
Jan

**Comment by lperron** [comment, marked as answer]
It cannot be, because floats are not consistent across platforms.
